lib/pbio/drv/usb/stm32_usbd/usbd_pybricks.c

```diff
@@ -0,0 +1,515 @@
+  * @{
+  */
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_Defines
+  * @{
+  */
+
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_Macros
+  * @{
+  */
+
+/**
+  * @}
+  */
+
+
+/** @defgroup USBD_Pybricks_Private_FunctionPrototypes
+  * @{
+  */
+
+static uint8_t USBD_Pybricks_Init(USBD_HandleTypeDef *pdev, uint8_t cfgidx);
+static uint8_t USBD_Pybricks_DeInit(USBD_HandleTypeDef *pdev, uint8_t cfgidx);
+static uint8_t USBD_Pybricks_Setup(USBD_HandleTypeDef *pdev, USBD_SetupReqTypedef *req);
+static uint8_t USBD_Pybricks_DataIn(USBD_HandleTypeDef *pdev, uint8_t epnum);
+static uint8_t USBD_Pybricks_DataOut(USBD_HandleTypeDef *pdev, uint8_t epnum);
+static uint8_t USBD_Pybricks_EP0_RxReady(USBD_HandleTypeDef *pdev);
+static uint8_t *USBD_Pybricks_GetCfgDesc(uint16_t *length);
+
+/**
+  * @}
+  */
+
+/** @defgroup USBD_Pybricks_Private_Variables
+  * @{
+  */
+
+USBD_ClassTypeDef USBD_Pybricks_ClassDriver =
+{
+    .Init = USBD_Pybricks_Init,
+    .DeInit = USBD_Pybricks_DeInit,
+    .Setup = USBD_Pybricks_Setup,
+    .EP0_RxReady = USBD_Pybricks_EP0_RxReady,
+    .DataIn = USBD_Pybricks_DataIn,
+    .DataOut = USBD_Pybricks_DataOut,
+    .GetFSConfigDescriptor = USBD_Pybricks_GetCfgDesc,
+};
+
+/* USB Pybricks device Configuration Descriptor */
+__ALIGN_BEGIN static uint8_t USBD_Pybricks_CfgDesc[USBD_PYBRICKS_CONFIG_DESC_SIZ] __ALIGN_END =
+{
+    /* Configuration Descriptor */
+    0x09,                                     /* bLength: Configuration Descriptor size */
+    USB_DESC_TYPE_CONFIGURATION,              /* bDescriptorType: Configuration */
+    LOBYTE(USBD_PYBRICKS_CONFIG_DESC_SIZ),    /* wTotalLength:no of returned bytes */
+    HIBYTE(USBD_PYBRICKS_CONFIG_DESC_SIZ),
+    0x01,                                     /* bNumInterfaces: 1 interface */
+    0x01,                                     /* bConfigurationValue: Configuration value */
+    0x00,                                     /* iConfiguration: Index of string descriptor describing the configuration */
+    0xC0,                                     /* bmAttributes: self powered */
+    0x32,                                     /* MaxPower 0 mA */
+
+    /*---------------------------------------------------------------------------*/
+
+    /* Data class interface descriptor */
+    0x09,                                     /* bLength: Endpoint Descriptor size */
+    USB_DESC_TYPE_INTERFACE,                  /* bDescriptorType: */
+    0x00,                                     /* bInterfaceNumber: Number of Interface */
+    0x00,                                     /* bAlternateSetting: Alternate setting */
+    0x02,                                     /* bNumEndpoints: Two endpoints used */
+    0xFF,                                     /* bInterfaceClass: vendor */
+    0x00,                                     /* bInterfaceSubClass: */
+    0x00,                                     /* bInterfaceProtocol: */
```

Comment:
I think we will want to make these the same as the device subclass and protocol. WinRT doesn't make the device subclass and protocol searchable but it does provide the values from the first interface.